app/controllers/course_summaries_controller.rb

```diff
@@ -2,17 +2,50 @@ class CourseSummariesController < ApplicationController
   include CourseSummariesHelper
 
   before_action :authorize_only_for_admin,
-                except: [:populate]
+                except: [:populate, :index]
 
   layout 'assignment_content'
 
   def index
   end
 
   def populate
+    averages = {}
+    medians = {}
+    visible_assessment_totals = []
+    if current_user.admin?
+      visible_assessments = Assessment.all.order(id: :asc).pluck_to_hash(:id, :type, :short_identifier)
+      MarkingScheme.all.each do |m|
+        averages[m.name] = DescriptiveStatistics.mean(m.students_weighted_grades_array(current_user)).round(2)
```

Comment:
This is good, but also add a median as well (since this will show up for admins). Make sure to save `m.students_weighted_grades_array(current_user)` in a variable to avoid calling the method twice.